server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -562,9 +560,9 @@ private List readArrayList() throws IOException {
         return list;
     }
 
-    private DateTime readDateTime() throws IOException {
+    private ZonedDateTime readDateTime() throws IOException {
         final String timeZoneId = readString();
-        return new DateTime(readLong(), DateTimeZone.forID(timeZoneId));
+        return ZonedDateTime.ofInstant(Instant.ofEpochMilli(readLong()), ZoneId.of(timeZoneId));
```

Comment:
Does `ZoneId.of` work for a larger range of values than `DateTimeZone.forID`? Or maybe we tell people not to use the ones that used to work and now don't?

Reply:
indeed, there are such cases, see https://github.com/elastic/elasticsearch/issues/27330#issuecomment-358958368

Reply:
Got it. So the plan is to add deprecation warnings to older versions of ES and not support them in master?